File: src/main/java/GreedyAlgorithm/GymClothes.java
package GreedyAlgorithm;

public class GymClothes {
    public static void main(String[] args) {
        int[] lost = {2, 4};
        int[] reserve = {1, 3, 5};

        int answer = solution(5, lost, reserve);
        System.out.println(answer);
    }

    public static int solution(int n, int[] lost, int[] reserve) {
        int count = n - lost.length;
        for (int i = 0; i < lost.length; i++) {
            for (int j = 0; j < reserve.length; j++) {
                if (lost[i] == reserve[j] + 1) {
                    count++;
                }
            }
        }
        return count;
    }
}

File: python/Greedy/gymClothes.py
def solution(n, lost, reserve):
    count = n - len(lost)
    arr = [False] * n

    for index in reserve:
        arr[index - 1] = True

    newLost = []
    # 체육복을 잃어버렸지만 여유분이 있을 때
    lost.sort()
    for index in lost:
        if arr[index - 1]:
            arr[index - 1] = False
            count += 1
        else:
            newLost.append(index - 1)

    for index in newLost:
        if index - 1 > -1 and arr[index - 1]:
            arr[index - 1] = False
            count += 1
        elif index + 1 < len(arr) and arr[index + 1]:
            arr[index + 1] = False
            count += 1
    print(count)


n = 5
lost = [4, 2]
reserve = [3, 5]

solution(n ,lost ,reserve)